LogIn/LoginViewController/LoginViewModel.swift

```diff
@@ -37,11 +40,20 @@ final class LoginViewModel: LoginViewModelType {
         }
     }
     
+    private lazy var completion: (Result<Bool, Error>) -> Void = { [weak self] result in
+        switch result {
+        case .success(let isLoggedIn):
+            self?.onLogin?()
+        case .failure(let error):
+            print(error.localizedDescription)
+        }
+    }
+    
     private func onLoginTap(email: String, passw: String) {
         if loginState == .login {
-            print("Log in", email, passw)
+            auth.signIn(email: email, password: passw, completion: completion)
         } else {
-            print("Sign up", email, passw)
+            auth.createUser(email: email, password: passw, completion: completion)
         }
     }
     
```

Comment:
Why do you need `emailString` and `passwString`?

Reply:
to pass it in login method